### ISSUE TITLE
SimpleQueryStringBuilderTests.testToQuery fails reproducibly on 6.8

### ISSUE THREAD

**Issue by gwbrown** [elastic/elasticsearch#54614]
[Build scan](https://gradle-enterprise.elastic.co/s/5x7i4fm5wrhc4/console-log?task=:server:unitTest)
[Jenkins](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+multijob+fast+part1/472/)

Per build-stats this has happened a couple times on 6.8, but not on any other branch.

Reproduce line (does reproduce locally):
```
./gradlew ':server:unitTest' -Dtests.seed=F39C7F6429B0DB86 -Dtests.class=org.elasticsearch.index.query.SimpleQueryStringBuilderTests -Dtests.method="testToQuery" -Dtests.security.manager=true -Dtests.locale=ar-LY -Dtests.timezone=America/Dawson -Dcompiler.java=12 -Druntime.java=8
```

Error:
```
FAILURE 1.66s | SimpleQueryStringBuilderTests.testToQuery <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: query was marked as cacheable in the context but this test indicates it should not be cacheable: {
   >   "simple_query_string" : {
   >     "query" : "NowWOr",
   >     "flags" : -1,
   >     "default_operator" : "or",
   >     "analyze_wildcard" : false,
   >     "quote_field_suffix" : "nqekmyhuhv",
   >     "minimum_should_match" : "1",
   >     "auto_generate_synonyms_phrase_query" : true,
   >     "fuzzy_prefix_length" : 5,
   >     "fuzzy_max_expansions" : 5,
   >     "fuzzy_transpositions" : true,
   >     "boost" : 0.25,
   >     "_name" : "XUSoaBCh4"
   >   }
   > }
   >    at __randomizedtesting.SeedInfo.seed([F39C7F6429B0DB86:4677D5A58331E6C]:0)
   >    at org.elasticsearch.test.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:433)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:567)
   >    at java.base/java.lang.Thread.run(Thread.java:835)
```

**Comment by cbuescher** [comment]
This fails due to is a rarely triggered edge case that was fixed on 8.0 and 7.2 by https://github.com/elastic/elasticsearch/pull/43199 that wasn't backported to 6.8. Will do that now.

**Comment by cbuescher** [comment]
Closed with https://github.com/elastic/elasticsearch/commit/acb4224d47cee6f8443bd9b2a33c99c3004189b9
